src/main/java/eu/dissco/orchestration/backend/repository/MachineAnnotationServiceRepository.java

```diff
@@ -0,0 +1,147 @@
+package eu.dissco.orchestration.backend.repository;
+
+import static eu.dissco.orchestration.backend.database.jooq.Tables.MACHINE_ANNOTATION_SERVICES;
+import static eu.dissco.orchestration.backend.repository.RepositoryUtils.getOffset;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import eu.dissco.orchestration.backend.database.jooq.tables.records.MachineAnnotationServicesRecord;
+import eu.dissco.orchestration.backend.domain.MachineAnnotationService;
+import eu.dissco.orchestration.backend.domain.MachineAnnotationServiceRecord;
+import eu.dissco.orchestration.backend.exception.DisscoJsonBMappingException;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.jooq.DSLContext;
+import org.jooq.JSONB;
+import org.springframework.stereotype.Repository;
+
+@Repository
+@RequiredArgsConstructor
+public class MachineAnnotationServiceRepository {
+
+  private final DSLContext context;
+  private final ObjectMapper mapper;
+
+  public void createMachineAnnotationService(MachineAnnotationServiceRecord masRecord) {
+    context.insertInto(MACHINE_ANNOTATION_SERVICES)
```

Comment:
If these properties are set to null, you'd push a null value in the repository. Is that the behaviour you want?

Reply:
Yes some values can be null, they are not really required.